### PR TITLE
Generate relation changes.

### PR DESCRIPTION
**Pull request by frankban**
Also send the correct record id for deploy changes.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/bundlechanges/12/
Test PASSed.

File: changes.go [hunk 1]
```diff
@@ -139,3 +158,29 @@ func handleRelations(add addChangeFunc, relations [][]string, addedServices map[
 func handleUnits(add addChangeFunc, services map[string]*charm.ServiceSpec, addedServices, addedMachines map[string]string) {
 	// TODO frankban: implement this.
 }
+
+// parseEndpoint creates an endpoint from its string representation.
+func parseEndpoint(e string) *endpoint {
+	parts := strings.Split(e, ":")
```
Comment (mhilton) [inline changes.go]: perhaps use strings.Splitn() ?

**Comment by mhilton** [comment]
LGTM

File: changes.go (hunk 1 above)
Comment (frankban) [inline changes.go]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/bundlechanges/13/
Test PASSed.

File: changes.go [hunk 2]
```diff
@@ -48,8 +49,14 @@ func (cs *changeset) add(method string, requires []string, args ...interface{})
 	if requires == nil {
 		requires = make([]string, 0)
 	}
+	// TODO frankban: current Python bundle lib includes this inconsistency;
+	// break compatibility and just use addService?
+	idPrefix := method
```
Comment (urosj) [inline changes.go]: Can we have versioned changes doc, so that bundlechanges and clients know what to expect?

Comment (urosj) [inline changes.go]: As per IRC discussion, this is handled on another layer.

File: changes.go [hunk 3]
```diff
@@ -139,3 +158,29 @@ func handleRelations(add addChangeFunc, relations [][]string, addedServices map[
 func handleUnits(add addChangeFunc, services map[string]*charm.ServiceSpec, addedServices, addedMachines map[string]string) {
 	// TODO frankban: implement this.
 }
+
+// parseEndpoint creates an endpoint from its string representation.
+func parseEndpoint(e string) *endpoint {
```
Comment (urosj) [inline changes.go]: What do you think about making this "resilient" to changes in the bundle format? Maybe this should go into Charm package instead?

**Comment by urosj** [comment]
:+1: With a suggestion.

File: changes.go (hunk 3 above)
Comment (frankban) [inline changes.go]: For now this is only intended as a helper for strings manipulation. But I agree with you that having shared code for endpoints would be useful.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/bundlechanges-merge

**Pull request merged**